memory-db/src/lib.rs

```diff
@@ -30,7 +30,7 @@ use std::{
 };
 
 #[cfg(not(feature = "std"))]
-use alloc::collections::btree_map::{BTreeMap as Map, Entry};
+use hashbrown::{hash_map::Entry, HashMap as Map};
```

Comment:
We can just use it on `std` and `no_std`. The `std` hashmap is actually hashbrown any way.